db/Pos/DB/Redirect.hs

```diff
@@ -48,3 +64,56 @@ instance
     dbDelete tag key = do
         db <- view (dbTagToLens tag) <$> getNodeDBs
         rocksDelete key db
+
+-- | Conduit source built from rocks iterator.
+iteratorSource ::
+       forall m i.
+       ( MonadRealDB m
+       , MonadResource m
+       , DBIteratorClass i
+       , Bi (IterKey i)
+       , Bi (IterValue i)
+       )
+    => DBTag
+    -> Proxy i
+    -> Source m (IterType i)
+iteratorSource tag _ = do
+    putText $ ("Iterator source, prefix " <> show (iterKeyPrefix @i))
+    DB{..} <- view (dbTagToLens tag) <$> lift getNodeDBs
+    let createIter = Rocks.createIter rocksDB rocksReadOpts
+    let releaseIter i = Rocks.releaseIter i
+    let onExc (e :: SomeException) = do
+            putText $ "Exception arised in redirect handler: " <> show e
```

Comment:
Moreover, it's usually considered a bad idea to catch exception, log it and throw further. I don't see a good reason why it's necessary here.